### ISSUE TITLE
Tests keep failing in headless mode

### ISSUE THREAD

**Issue by SDAdham** [cypress-io/cypress#17352]
### Current behavior

I am experiencing the same issue as https://github.com/cypress-io/cypress/issues/5098

What happens is that 
* I visit a page, 
* click a link that should route me to another page 
* then validate the url to should include the sub path

The steps above runs in browser, but randomly in local and CI (both), it fails at some point for those steps but always at "validate the url to should include the sub path" when after clicking the link is not really happening.

I check the screenshot produced and I can confirm the right link is clicked but it just simply looks like as if that link wasn't clicked.

The link is never disabled, always working and the steps are working just fine again in a browser mode.

### Desired behavior

It should always navigate after clicking a link

### Test code to reproduce

There is not really straight forward code to reproduce. As mentioned, sometimes it fails and sometimes not at different location of the same process.

```
cy.visit('url');
cy.contains('link text').click().wait(1000).url().should('include', 'sub-route')
```

### Cypress Version

7.7.0

### Other

In case anyone suggests that it might be my application, I reviewed it a lot, I can guarantee not, cuz in browser mode, it ALWAYS succeeds.

**Comment by JohnnyFun** [comment]
If you break the command into separate lines does it help? Like:

```
cy.visit('url');
cy.contains('link text').click()
cy.wait(1000)
cy.url().should('include', 'sub-route')
```

Also, I don't _think_ you need to wait(1000), since I'm pretty sure cypress will automatically check the url until it includes "sub-route" or until a few seconds have passed.

**Comment by SDAdham** [comment]
@JohnnyFun , no it doesn't help. I had to comment those lines, but it got stuck at another point with the same sequence.

**Comment by SDAdham** [comment]
You were right @JohnnyFun . Thanks, there was something wrong with the ci build.

**Comment by JohnnyFun** [comment]
What was the issue? Glad you got it working!

**Comment by SDAdham** [comment]
it turned out that ci build for UI went wrong since upgrade to Angular 12. The local dev build is working perfectly.

**Comment by JohnnyFun** [comment]
So was there a different version of angular was being run on CI for some reason? Was it an `npm` issue?

**Comment by SDAdham** [comment]
I upgraded my local angular from 10 to 12, then I adjusted a lot of configurations of which one of them is `angular.json`, apparently, I forgot to update the `development/production` configurations there which `ci` uses. In my local I use angular `serve` which is slightly different.

Cypress was right, if it does click a link, it wouldn't navigate to the route cuz the route wasn't working cuz of a hashing issue for the `development/production`. Once I fixed that, it all worked.
